File: rmt/src/test/scala/de/flower/test/ExceptionLoggerTestListener.scala
package de.flower.test

import org.slf4j.{LoggerFactory, Logger}
import org.testng.{Assert, TestListenerAdapter, ITestResult}

/**
 *
 * @author flowerrrr
 */

class ExceptionLoggerTestListener extends TestListenerAdapter {

    private final val log: Logger = LoggerFactory.getLogger(getClass)

    override def onTestFailure(tr: ITestResult): Unit = {
        log.error(tr.toString, tr.getThrowable)
    }

    override def onTestSkipped(tr: ITestResult): Unit = {
        log.error(tr.toString, tr.getThrowable)
    }

}

File: rmt/src/test/java/de/flower/common/test/ExceptionLoggerTestListener.java
package de.flower.common.test;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.testng.ITestResult;
import org.testng.TestListenerAdapter;



public class ExceptionLoggerTestListener extends TestListenerAdapter {

    private final Logger log = LoggerFactory.getLogger(getClass());

    @Override
    public void onTestFailure(ITestResult tr) {
        log.error(tr.toString(), tr.getThrowable());
    }

    @Override
    public void onTestSkipped(ITestResult tr) {
        log.error(tr.toString(), tr.getThrowable());
    }
}